code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -433,36 +499,51 @@ to_chat will check for valid clients itself already so no need to double check f
 
 
 /datum/hive_status/normal/proc/can_spawn_larva(mob/xeno_candidate)
+	if(!xeno_candidate?.client)
+		return FALSE
+
 	if(!stored_larva)
 		to_chat(xeno_candidate, "<span class='warning'>There are no burrowed larvas.</span>")
 		return FALSE
 
-	if(!living_xeno_queen?.ovipositor)
-		to_chat(xeno_candidate, "<span class='warning'>There are no mothers with an ovipositor deployed.</span>")
+	var/list/possible_mothers = list()
+	if(living_xeno_queen?.is_burrowed_larva_host())
+		possible_mothers += living_xeno_queen
+	if(living_xeno_shrike?.is_burrowed_larva_host())
+		possible_mothers += living_xeno_shrike
+
+	if(!length(possible_mothers))
+		to_chat(xeno_candidate, "<span class='warning'>There are no mothers currently available to recive new larvas.</span>")
 		return FALSE
 
-	if (!xeno_candidate || !xeno_candidate.client)
+	var/mob/living/carbon/xenomorph/chosen_mother = input("Available Mothers") as null|anything in possible_mothers
+	if(QDELETED(chosen_mother) || !xeno_candidate?.client)
 		return FALSE
 
-	if(living_xeno_queen?.incapacitated(TRUE))
-		to_chat(xeno_candidate, "<span class='warning'>Mother is not in a state to receive us.</span>")
+	if(!chosen_mother?.is_burrowed_larva_host())
+		to_chat(xeno_candidate, "<span class='warning'>This mother is no longer in a state to receive us.</span>")
 		return FALSE
 
 	if(!isnewplayer(xeno_candidate) && !DEATHTIME_CHECK(xeno_candidate))
 		DEATHTIME_MESSAGE(xeno_candidate)
 		return FALSE
 
-	return TRUE
+	return chosen_mother
```

Comment:
this is can_spawn_larva not get_queen_or_shrike_to_spawn_from

Reply:
If there's no queen nor shrike to spawn from then it cannot spawn a larva, coincidentally.
And if it can, well, a reference to a mob is a true value.